src/closures.rs

```diff
@@ -273,10 +285,29 @@ fn rewrite_closure_fn_decl(
         .tactic(tactic)
         .preserve_newline(true);
     let list_str = write_list(&item_vec, &fmt)?;
-    let mut prefix = format!("{}{}{}|{}|", is_async, immovable, mover, list_str);
+    let one_line_budget = context.budget(param_shape.indent.width());
+    let (param_str, put_params_in_block) = if match indent_style {
```

Comment:
Would you mind factoring this `match` expression to a variable to make this if condition simpler?

Reply:
Good idea, will do